load.go

```diff
@@ -177,3 +181,14 @@ func readFile(templateKey string, baseDir string, filePath string) string {
 	return string(dat)
 
 }
+
+func (m Mock) patchedWith(patch Mock) *Mock {
```

Comment:
Since it introduces side effect, better just do
```
func (m *Mock) patchedWith(patch Mock) {
  ...
}
```

Reply:
i wanted to avoid mutating the original parent mock, this returns a copy of it, but with updates from the patch. is the pointer receiver type still better if that's my intent?

Reply:
I see, makes sense, can you add a unit test that makes sure the original m doesn't change?

Reply:
done!